src/rdb.c

```diff
@@ -1396,9 +1397,22 @@ ssize_t rdbSaveDb(rio *rdb, int dbid, int rdbflags, long *key_counter) {
 
     kvs_it = kvstoreIteratorInit(db->keys);
     int last_slot = -1;
+    clusterNode *myNode = NULL;
+    if (server.cluster_enabled) {
+        myNode = getMyClusterNode();
+        if (NULL == myNode) {
```

Comment:
is there a chance it can be null here?

Reply:
Yeah, this can be null, depending on the cluster API implementation. Indeed, the implementation in cluster_legacy.c does not allow this, but with a different cluster implementation, it can theoretically happen.

Reply:
a style problem, we usually  use` if (myNode == NULL)` (the compiler can detect confusion between `==` and `=`), and local variables are usually written in lowercase, for example, `my_node`